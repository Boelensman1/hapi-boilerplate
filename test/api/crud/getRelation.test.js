const path = require('path')
const setUpCrudTest = require('test/api/setUpCrudTest')
const { user } = require('models')

const seedLocation = path.join(__dirname, 'getRelation.seed.yaml')

const { getRelation } = require('api/crud')

describe('Test getRelation crud function', () => {
  test('Get relation', async () => {
    const { request, h } = await setUpCrudTest(seedLocation)

    request.params = { id: 10 }
    const handler = getRelation('role', 'user', user).handler.bind(
      null,
      request,
      h,
    )
    const {
      body: { result },
    } = await handler()

    expect(result).toHaveLength(1)
    expect(result[0]).toMatchSnapshot({
      createdAt: expect.any(Date),
    })
  })
})
