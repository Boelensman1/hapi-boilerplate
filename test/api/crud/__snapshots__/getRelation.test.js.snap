// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test getRelation crud function Get relation 1`] = `
Object {
  "createdAt": Any<Date>,
  "id": 100,
  "loggedInAt": null,
  "roleId": 10,
  "updatedAt": null,
  "username": "Relation User",
}
`;
